helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/StoppableInstancesSelector.java

```diff
@@ -99,13 +121,15 @@ public void getStoppableInstancesCrossZones(List<String> instances,
       if (instanceSet.isEmpty()) {
         continue;
       }
-      getStoppableInstances(new ArrayList<>(instanceSet), toBeStoppedInstances);
+      getStoppableInstances(new ArrayList<>(instanceSet), toBeStoppedInstancesSet, stoppableInstances,
+          failedStoppableInstances);
     }
-    processNonexistentInstances(instances);
+    processNonexistentInstances(instances, failedStoppableInstances);
+    return result;
   }
 
-  private void getStoppableInstances(List<String> instances, Set<String> toBeStoppedInstances)
-      throws IOException {
+  private void getStoppableInstances(List<String> instances, Set<String> toBeStoppedInstances,
```

Comment:
nit: I would call this `populateStoppableInstances` or something along those lines since this method has side effect of modifying a list that is passed by reference.